### PR TITLE
Add theory field to test topic model. Add router and service method f…

### PR DESCRIPTION
**Pull request by lebedevaanastasiya**
…or updating test topic theory.

### CODE REVIEW

File: routes/testTopics/index.js [hunk 1]
```diff
@@ -77,4 +77,27 @@ router.delete('/:id', (req, res, next) => {
         });
 });
 
+router.put('/:id/theory', (req, res, next) => {
+    const id = req.params.id || null;
+    const theory = req.body.theory || null;
+    if(!(id && theory)) {
+        res.status(400);
```
Comment (mbarouski) [inline routes/testTopics/index.js]: !id || !theory

Comment (mbarouski) [reply routes/testTopics/index.js]: try to use more readable if statements

File: routes/testTopics/index.js [hunk 2]
```diff
@@ -77,4 +77,27 @@ router.delete('/:id', (req, res, next) => {
...
+    testTopicsService.updateTheory(id, theory)
+        .then(data => {
+            res.json({
+                data: data
+            })
+        })
+        .catch(err => {
+            res.status(err.status);
+            res.json({
+                data: err.message
```
Comment (mbarouski) [inline routes/testTopics/index.js:98]: do we send error msg in data field?

File: services/testTopics/index.js [hunk 3]
```diff
@@ -38,9 +38,22 @@ function remove(id) {
     });
 }
 
+function updateTheory(id, theory) {
+    return new Promise((resolve, reject) => {
+        TestTopic.update({ theory }, { where: { id } })
+            .then(result => {
+                TestTopic.findOne({ where: { id } })
+                    .then( topic => resolve(topic))
+                    .catch( err => reject({ status: 500 }));
```
Comment (mbarouski) [inline services/testTopics/index.js:47]: it's incorrect, because if we can't find object with id, it's not 500, we have no such record in db simple

Comment (mbarouski) [reply services/testTopics/index.js:47]: we can't send 500 in all cases

**Review by mbarouski — CHANGES_REQUESTED** [review]

File: routes/testTopics/index.js (hunk 2 above)
Comment (lebedevaanastasiya) [reply routes/testTopics/index.js:98]: Yes, we do

File: services/testTopics/index.js (hunk 3 above)
Comment (lebedevaanastasiya) [reply services/testTopics/index.js:47]: What status in this case should we send?

Comment (mbarouski) [reply services/testTopics/index.js:47]: different statuses

Comment (lebedevaanastasiya) [reply services/testTopics/index.js:47]: How can I now that different return errors and handle them?

Comment (mbarouski) [reply services/testTopics/index.js:47]: so,you can not change smth in this project, but we need ti analyze exception object in catch block

File: routes/testTopics/index.js [hunk 4]
```diff
@@ -77,4 +77,27 @@ router.delete('/:id', (req, res, next) => {
         });
 });
 
+router.put('/:id/theory', (req, res, next) => {
+    const id = req.params.id || null;
+    const theory = req.body.theory || null;
+    if(!id && !theory) {
```
Comment (mbarouski) [inline routes/testTopics/index.js]: ||

File: routes/testResults/index.js [hunk 5]
```diff
@@ -28,7 +28,7 @@ router.get('/:userId', (req, res, next) => {
 router.get('/:userId/:testId', (req, res, next) => {
    const userId = req.params.userId || null;
    const testId = req.params.testId || null;
-   if(!(userId && (userId > 0) && testId && (testId > 0))){
+   if(!userId && !testId){
```
Comment (mbarouski) [inline routes/testResults/index.js]: ||

**Review by mbarouski — CHANGES_REQUESTED** [review]

**Pull request merged**